### PR TITLE
Implement PsAll

### LINKED ISSUE #350: PsAll is missing
Takes misses `Pass` implementation that will require each provided `Pass` to success in order to the target `Pass` to success. 
The concept is similar to python [all](https://docs.python.org/2/library/functions.html#all) builtin.

The contract could be like this:
```java
public PsAll(final Iterable<Pass> list, final int index)
```

The `index` parameter will indicate which `Identity` to return once all suceeded.

### PR DESCRIPTION
**Pull request by Suseika**
Implement `PsAll`. This class is a `Pass` implementatoin that combines multiple `Pass`es. It can be entered if every underlying `Pass` can be entered.

Fixes #350

### CODE REVIEW

**Comment by Suseika** [comment]
@davvd Pease assign someone to review this PR

**Comment by davvd** [comment]
@suseika I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by davvd** [comment]
@dmzaytsev it's yours,please review

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            new PsAll(
+                Arrays.asList(
+                    new PsFake(true),
+                    new PsFake(true),
+                    new PsFake(true),
+                    new PsFake(true)
+                ),
+                Tv.THREE
+            ).enter(new RqFake()).has(),
+            Matchers.is(true)
```
Comment (dmzaytsev) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @Suseika how about to check PsAll.enter() returns `Identity` from 3-rd `Pass`?

File: src/main/java/org/takes/facets/auth/PsAll.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        if (this.allMatch(request)) {
+            result = this.all.get(this.index).enter(request);
+        } else {
+            result = new Opt.Empty<Identity>();
+        }
+        return result;
+    }
+
+    @Override
+    public final Response exit(final Response response, final Identity identity)
```
Comment (dmzaytsev) [inline src/main/java/org/takes/facets/auth/PsAll.java]: @Suseika please add some tests for this method

File: src/main/java/org/takes/facets/auth/PsAll.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    private final transient int index;
+
+    /**
+     * Ctor.
+     * @param passes All Passes to be checked.
+     * @param identity Index of a Pass whose Identity to return on successful
+     *  {@link PsAll#enter(Request)}
+     */
+    public PsAll(final List<? extends Pass> passes, final int identity) {
+        this.all = passes;
```
Comment (dmzaytsev) [inline src/main/java/org/takes/facets/auth/PsAll.java]: @Suseika let's clone the list instead assignment

**Comment by dmzaytsev** [comment]
@Suseika 3 comments above

**Comment by Suseika** [comment]
@dmzaytsev I fixed what you mentioned in the comments, please review.

File: src/test/java/org/takes/facets/auth/PsAllTest.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            new PsAll(
+                Arrays.asList(
+                    new PsFake(true),
+                    new PsFake(true),
+                    new PsFake(true),
+                    resulting
+                ),
+                Tv.THREE
+            ).enter(request).get().urn(),
+            Matchers.equalTo(resulting.enter(request).get().urn())
```
Comment (dmzaytsev) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @Suseika are you sure? try replace `Tv.THREE` to `Tv.TWO` I think you will get same result.
I would suggest to use `PsFixed` instead  `PsFake` for `resulting`

**Comment by dmzaytsev** [comment]
@Suseika one comment above

File: src/test/java/org/takes/facets/auth/PsAllTest.java (hunk 4 above)
Comment (Suseika) [inline src/test/java/org/takes/facets/auth/PsAllTest.java]: @dmzaytsev Thanks for the suggestion.

**Comment by Suseika** [comment]
@dmzaytsev Done.

**Comment by dmzaytsev** [comment]
@Suseika  thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@Suseika the code looks cool, thanks. but why did you change `logo.png` file?

**Comment by Suseika** [comment]
@yegor256 Its content changed somehow and I was unable to revert it after several attempts, and then forgot to exclude it from a commit. Maybe that was automatic CRLF substitution? Or something within my IDE. I'm not sure. I'll try to fix that.

**Comment by Suseika** [comment]
@yegor256 I removed a commit which had the `logo.png` changed in my fork. Now it's good to merge, I guess. But I still don't know why I couldn't revert that file, even with `git checkout -- src/site/resources/logo.png` it remained modified in `git status`. 

`logo.png` probably contains CRLF line endings mixed with LFs (I'll check that later, not sure how to do that right now), and our `.gitattributes` forces line endings to be LF. But then other developers should have faced that same issue.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@ypshenychka review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by Suseika** [comment]
@yegor256 Build is failing, and it seems unrelated to source changes.

**Comment by ypshenychka** [comment]
@davvd Quality is good here.

**Comment by davvd** [comment]
> @davvd Quality is good here.

@ypshenychka thanks a lot ;)

**Comment by davvd** [comment]
@dmzaytsev I added 10 mins to @ypshenychka (for QA review) in transaction 61670586; Thank you, I have added **19 mins** to your account in payment/transaction "`61670623`", time consumed: 23 hours and 36 mins; you're getting extra minutes here (c=4); +19 added to your rating, at the moment it is: [+2221](http://www.netbout.com/b/35166?open=rating)

**Pull request merged**